src/Dialect/ONNX/Transforms/Decompose.cpp

```diff
@@ -668,37 +669,524 @@ bool shouldDecomposeConvTransposeOp(Value convTransposeResult) {
+      auto startOnnxConstant = getONNXConstOpFromVector(startVector);
+      auto endOnnxConstant = getONNXConstOpFromVector(endVector);
+      weightSlices.push_back(
+          create.onnx.slice(convWeightsShapedType, weights, startOnnxConstant,
+              endOnnxConstant, axisOnnxConstant, stepOnnxConstant));
+    }
+  }
+  auto convKernelShapeArrayAttr =
+      rewriter.getI64ArrayAttr({convKernelSize, convKernelSize});
+  // This is the shape of the output from each conv, which contributes to the
+  // final ofm.
+  SmallVector<int64_t> convOutputShape(convTransposeOutputShape);
+  convOutputShape[convOutputShape.size() - 1] =
+      convOutputShape[convOutputShape.size() - 1] / stridesShape[0];
+  convOutputShape[convOutputShape.size() - 2] =
+      convOutputShape[convOutputShape.size() - 2] / stridesShape[0];
+  ShapedType convTransposeOutputType =
+      mlir::cast<ShapedType>(op.getY().getType());
+  // In the case where weights are padded, we will get the extra output from
+  // conv.
+  auto convOutputType = RankedTensorType::get(
+      (shouldPadWeights)
+          ? SmallVector<int64_t>({convOutputShape[0], convOutputShape[1],
+                convOutputShape[2] + 1, convOutputShape[3] + 1})
+          : convOutputShape,
+      convTransposeOutputType.getElementType());
+  if (numPhases == 4) {
+    auto padsArrayAttr = rewriter.getI64ArrayAttr({1, 1, 1, 1});
+    auto stridesArrayAttr = rewriter.getI64ArrayAttr({1, 1});
+
+    Value conv1 = rewriter.create<ONNXConvOp>(loc, convOutputType, input,
+        weightSlices[3], bias, mlir::StringAttr(), dilations, group,
+        convKernelShapeArrayAttr, padsArrayAttr, stridesArrayAttr);
+    Value conv2 = rewriter.create<ONNXConvOp>(loc, convOutputType, input,
+        weightSlices[0], bias, mlir::StringAttr(), dilations, group,
+        convKernelShapeArrayAttr, padsArrayAttr, stridesArrayAttr);
+    Value conv3 = rewriter.create<ONNXConvOp>(loc, convOutputType, input,
+        weightSlices[1], bias, mlir::StringAttr(), dilations, group,
+        convKernelShapeArrayAttr, padsArrayAttr, stridesArrayAttr);
+    Value conv4 = rewriter.create<ONNXConvOp>(loc, convOutputType, input,
+        weightSlices[2], bias, mlir::StringAttr(), dilations, group,
+        convKernelShapeArrayAttr, padsArrayAttr, stridesArrayAttr);
+
+    // Need to remove excess the ofm  when weights are padded.
+    if (shouldPadWeights) {
+      auto startOnnxConstant = getONNXConstOpFromVector({1, 1});
+      auto endOnnxConstant = getONNXConstOpFromVector(
+          {convOutputShape[convOutputShape.size() - 2] + 2,
+              convOutputShape[convOutputShape.size() - 1] + 2});
+      auto axisOnnxConstant = getONNXConstOpFromVector({2, 3});
+      auto stepOnnxConstant = getONNXConstOpFromVector({1, 1});
+      auto convSliceOutputType = RankedTensorType::get(
+          llvm::ArrayRef({convOutputShape[0], convOutputShape[1],
+              convOutputShape[2], convOutputShape[3]}),
```

Comment:
```suggestion
      auto convSliceOutputType = RankedTensorType::get(convOutputShape),
```
I do not think the manual conversion is needed here

Reply:
Correct, fixed it.